extensions/amp-date-picker/0.1/amp-date-picker.js

```diff
@@ -228,67 +392,430 @@ class AmpDatePicker extends AMP.BaseElement {
+  setState_(newState) {
+    return this.render(Object.assign(this.state_, newState));
+  }
+
+  /**
+   * Get the existing input, or create a hidden input for the date field.
+   * @param {!DateFieldType} type The selector for the input field
+   * @return {?Element}
+   */
+  setupDateField_(type) {
+    const fieldSelector = this.element.getAttribute(`${type}-field`);
+    const existingField = this.ampdoc_.getRootNode().querySelector(
+        fieldSelector);
+    const form = this.element.closest('form');
+
+    if (existingField) {
+      return existingField;
+    } else if (this.mode_ == DatePickerMode.STATIC && form) {
+      const hiddenInput = this.document_.createElement('input');
+      hiddenInput.type = 'hidden';
+      hiddenInput.name = this.getHiddenInputId(form, type);
+      this.element.appendChild(hiddenInput);
+      return hiddenInput;
     } else {
       return null;
     }
   }
 
   /**
-   * Create an array of objects mapping dates to templates.
-   * @param {?Promise<!JsonObject|!Array<JsonObject>>} srcTemplatePromise
-   * @return {!Promise<undefined>}
+   * Generate a name for a hidden input.
+   * Date pickers not in a form don't need named hidden inputs.
+   * @param {!Element} form
+   * @param {!DateFieldType} type
+   */
+  getHiddenInputId(form, type) {
+    const id = this.element.id;
+    if (form) {
+      const alternativeName = `${id}-${type}`;
+      if (!form.elements[type]) {
+        return type;
+      } else if (id && !form.elements[alternativeName]) {
+        return `${id}-${type}`;
+      } else {
+        user().error(TAG,
+            `Multiple date-pickers with implicit ${type} fields` +
+            'need to have IDs');
+        return '';
+      }
+    }
+  }
+
+  /**
+   * Add the document-level event listeners to attach behavior to any referenced
+   * input fields and to enable overlay open/close.
+   */
+  setupListeners_() {
+    const documentElement = this.element.ownerDocument.documentElement;
+    // Only add for overlay since click events just handle opening and closing
+    if (this.mode_ == DatePickerMode.OVERLAY) {
+      this.listen_(documentElement, 'click', this.handleClick_.bind(this));
+    }
+
+    this.listen_(documentElement, 'focusin', this.handleFocus_.bind(this));
+    this.listen_(documentElement, 'input', this.handleInput_.bind(this));
+    this.listen_(documentElement, 'keydown', this.handleKeydown_.bind(this));
+  }
+
+  /**
+   * True if the input is a field of this date picker.
+   * @param {?Element} field
+   * @return {boolean}
+   */
+  isDateField_(field) {
+    return (
+      field === this.dateField_ ||
+      field === this.startDateField_ ||
+      field === this.endDateField_
+    );
+  }
+
+  /**
+   * Handle clicks inside and outside of the date picker to detect when to
+   * open and close the date picker.
+   * @param {!Event} e
+   */
+  handleClick_(e) {
+    // Disregard spacebar and enter "clicks"
+    if (e.screenX == 0 && e.screenY == 0) {
```

Comment:
won't `this.isDateField_(target)` handle this anyway?

Reply:
Looks like it, thanks